[WIP] Adding b-tensor encoding scripts like compute_fodf/frf and metrics

karanphil opened the pull request:
Here are three scripts `scil_compute_memsmt_frf.py`, `scil_compute_memsmt_fofd.py` and `scil_compute_divide.py` enabling the use of tensor-valued dMRI data. The scripts are almost Scilpy ready, but the side modules are messy. This needs to be cleaned up before the PR is ready.

pep8speaks commented:
Hello @karanphil, Thank you for submitting the Pull Request !

* In the file [`scilpy/reconst/b_tensor_utils.py`](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py):

> [Line 30:19](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L30): [E711](https://duckduckgo.com/?q=pep8%20E711) comparison to None should be 'if cond is not None:'
> [Line 54:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L54): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 55:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L55): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 59:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L59): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 66:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L66): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 68:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L68): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 69:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L69): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 70:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L70): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 71:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L71): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 72:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L72): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (116 > 79 characters)
> [Line 74:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L74): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 75:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L75): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 80:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L80): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 95:45](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L95): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 95:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L95): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 96:45](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L96): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 107:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L107): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (99 > 79 characters)
> [Line 109:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L109): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 113:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L113): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 123:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L123): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 135:19](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L135): [E711](https://duckduckgo.com/?q=pep8%20E711) comparison to None should be 'if cond is not None:'
> [Line 155:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L155): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 163:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L163): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 182:5](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L182): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 183:5](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L183): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 184:5](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L184): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 184:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L184): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 191:23](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/b_tensor_utils.py#L191): [E711](https://duckduckgo.com/?q=pep8%20E711) comparison to None should be 'if cond is not None:'

* In the file [`scilpy/reconst/divide_fit.py`](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py):

> [Line 16:59](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L16): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 29:5](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L29): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 33:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L33): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (159 > 79 characters)
> [Line 35:52](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L35): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 35:52](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L35): [E712](https://duckduckgo.com/?q=pep8%20E712) comparison to True should be 'if cond is True:' or 'if cond:'
> [Line 39:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L39): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 78:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L78): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 79:9](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L79): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 81:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L81): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 79 characters)
> [Line 82:37](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L82): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 82:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L82): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 82:103](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L82): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 82:103](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L82): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 83:9](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L83): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 89:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L89): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 90:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L90): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (145 > 79 characters)
> [Line 91:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L91): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 108:20](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L108): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 111:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L111): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 129:20](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L129): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 130:20](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L130): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 131:21](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L131): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 132:21](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L132): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 136:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L136): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 137:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L137): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 138:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L138): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 141:37](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/divide_fit.py#L141): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

* In the file [`scilpy/reconst/multi_processes.py`](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/multi_processes.py):

> [Line 560:21](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scilpy/reconst/multi_processes.py#L560): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

* In the file [`scripts/scil_compute_divide.py`](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py):

> [Line 25:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L25): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 137:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L137): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 138:33](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L138): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 138:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L138): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (142 > 79 characters)
> [Line 165:29](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L165): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 167:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L167): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 168:57](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L168): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 169:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L169): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 181:25](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L181): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 194:5](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L194): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 195:5](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L195): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 198:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L198): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 199:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L199): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 200:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L200): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 201:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L201): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 202:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L202): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 202:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L202): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 203:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L203): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 204:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L204): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 212:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L212): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 218:50](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L218): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 230:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_divide.py#L230): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1

* In the file [`scripts/scil_compute_memsmt_fodf.py`](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py):

> [Line 31:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L31): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 128:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L128): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 145:43](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L145): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 184:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L184): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 188:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L188): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 193:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L193): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 196:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L196): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 208:33](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L208): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 208:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L208): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (142 > 79 characters)
> [Line 234:29](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L234): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 236:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L236): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 237:57](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L237): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 238:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L238): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 245:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L245): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 299:40](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L299): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 300:40](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L300): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 304:31](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L304): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 313:37](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L313): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 322:37](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L322): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 323:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L323): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 331:37](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L331): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 334:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_fodf.py#L334): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)

* In the file [`scripts/scil_compute_memsmt_frf.py`](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py):

> [Line 5:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L5): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (123 > 79 characters)
> [Line 37:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L37): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 40:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L40): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 111:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L111): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 121:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L121): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 131:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L131): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 138:69](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L138): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 190:21](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L190): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 191:33](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L191): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 192:33](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L192): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 193:33](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L193): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 194:33](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L194): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 195:33](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L195): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 201:29](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L201): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 203:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L203): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 204:57](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L204): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 205:28](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L205): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 212:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L212): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 227:80](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L227): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 227:91](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L227): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 237:53](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L237): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 238:53](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L238): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 239:53](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L239): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 240:53](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L240): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 241:53](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L241): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 242:53](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L242): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 243:53](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L243): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 294:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L294): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 308:1](https://github.com/scilus/scilpy/blob/d56e04a8a486c8c93ee0b9237ed2a60a48e8abca/scripts/scil_compute_memsmt_frf.py#L308): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/8/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/10/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/11/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/12/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/13/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/14/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/15/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/16/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/17/

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

karanphil commented:
@arnaudbore if you want to test the three scripts, here is some data.

https://usherbrooke-my.sharepoint.com/:f:/g/personal/karp2601_usherbrooke_ca/EqLMY8wcU85EjGT4J19kAhABBgsdVS3EF3QgVsdDlG23jA?e=TTlnYQ

Here are examples of command lines for the scripts.

`scil_compute_memsmt_frf.py wm_frf.txt gm_frf.txt csf_frf.txt --in_dwi_linear dwi_lin_eddy_corrected_n4.nii.gz --in_bval_linear dwi_lin.bval --in_bvec_linear dwi_lin.bvec --in_dwi_spherical dwi_sph_eddy_corrected_n4.nii.gz --in_bval_spherical dwi_sph.bval --in_bvec_spherical dwi_sph.bvec --mask mask.nii.gz --mask_wm wm_mask.nii.gz --mask_gm gm_mask.nii.gz --mask_csf csf_mask.nii.gz --roi_center 45 51 29 --wm_frf_mask wm_frf_mask.nii.gz --gm_frf_mask gm_frf_mask.nii.gz --csf_frf_mask csf_frf_mask.nii.gz -f`

`scil_compute_memsmt_fodf.py wm_frf.txt gm_frf.txt csf_frf.txt --in_dwi_linear dwi_lin_eddy_corrected_n4.nii.gz --in_bval_linear dwi_lin.bval --in_bvec_linear dwi_lin.bvec --in_dwi_spherical dwi_sph_eddy_corrected_n4.nii.gz --in_bval_spherical dwi_sph.bval --in_bvec_spherical dwi_sph.bvec --mask mask.nii.gz --processes 8 -f`

`scil_compute_divide.py --in_dwi_linear dwi_lin_eddy_corrected_n4.nii.gz --in_bval_linear dwi_lin.bval --in_bvec_linear dwi_lin.bvec --in_dwi_spherical dwi_sph_eddy_corrected_n4.nii.gz --in_bval_spherical dwi_sph.bval --in_bvec_spherical dwi_sph.bvec --mask mask.nii.gz --fa fa.nii.gz  -f`

mdesco reviewed (commented):
I have launched scil_compute_divide.py with only linear data it ran... I don't think it should @karanphil , no?

Otherwise, @arnaudbore we approve for now and move my small comments to issues that @karanphil  can correct when he gets back

mdesco commented:
@karanphil what is the status on this PR? Welcome back! :)

karanphil commented:
> @karanphil what is the status on this PR? Welcome back! :)

Thanks! I'm looking into it. I'll come back to you when it's ready!

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/20/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/21/

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/29/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/30/

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/32/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/33/

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/38/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/39/

karanphil commented:
@arnaudbore @mdesco I cleaned up the code and changed a few things. I think it's ready for testing. Here is some data.

https://usherbrooke-my.sharepoint.com/:f:/g/personal/karp2601_usherbrooke_ca/EqLMY8wcU85EjGT4J19kAhABBgsdVS3EF3QgVsdDlG23jA?e=TTlnYQ

Here are examples of command lines for the scripts.

`scil_compute_memsmt_frf.py wm_frf.txt gm_frf.txt csf_frf.txt --in_dwis dwi_lin_eddy_corrected_n4.nii.gz dwi_sph_eddy_corrected_n4.nii.gz --in_bvals dwi_lin.bval dwi_sph.bval --in_bvecs dwi_lin.bvec dwi_sph.bvec --in_bdeltas 1 0 --mask mask.nii.gz --mask_wm wm_mask.nii.gz --mask_gm gm_mask.nii.gz --mask_csf csf_mask.nii.gz --roi_center 45 51 29 --wm_frf_mask wm_frf_mask.nii.gz --gm_frf_mask gm_frf_mask.nii.gz --csf_frf_mask csf_frf_mask.nii.gz -f`

`scil_compute_memsmt_fodf.py wm_frf.txt gm_frf.txt csf_frf.txt --in_dwis dwi_lin_eddy_corrected_n4.nii.gz dwi_sph_eddy_corrected_n4.nii.gz --in_bvals dwi_lin.bval dwi_sph.bval --in_bvecs dwi_lin.bvec dwi_sph.bvec --in_bdeltas 1 0 --mask mask.nii.gz --processes 8 -f`

`scil_compute_divide.py --in_dwis dwi_lin_eddy_corrected_n4.nii.gz dwi_sph_eddy_corrected_n4.nii.gz --in_bvals dwi_lin.bval dwi_sph.bval --in_bvecs dwi_lin.bvec dwi_sph.bvec --in_bdeltas 1 0 --mask mask.nii.gz --fa fa.nii.gz -f`

Thanks!

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build passed ! Good Job :beers: !

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/47/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/48/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/49/

scil-admin commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/50/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/51/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/52/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/53/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/54/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/55/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/56/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/57/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/58/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/59/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/60/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/61/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/62/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/63/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/64/

arnaudbore commented:
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-425/65/

arnaudbore commented:
Build passed ! Good Job :beers: !

karanphil commented:
> @arnaudbore @mdesco I cleaned up the code and changed a few things. I think it's ready for testing. Here is some data.
> 
> https://usherbrooke-my.sharepoint.com/:f:/g/personal/karp2601_usherbrooke_ca/EqLMY8wcU85EjGT4J19kAhABBgsdVS3EF3QgVsdDlG23jA?e=TTlnYQ
> 
> Here are examples of command lines for the scripts.
> 
> `scil_compute_memsmt_frf.py wm_frf.txt gm_frf.txt csf_frf.txt --in_dwis dwi_lin_eddy_corrected_n4.nii.gz dwi_sph_eddy_corrected_n4.nii.gz --in_bvals dwi_lin.bval dwi_sph.bval --in_bvecs dwi_lin_eddy_corrected.eddy_rotated_bvecs dwi_sph_eddy_corrected.eddy_rotated_bvecs --in_bdeltas 1 0 --mask mean_b0_lin_topup_bet_mask.nii.gz --mask_wm wm_mask.nii.gz --mask_gm gm_mask.nii.gz --mask_csf csf_mask.nii.gz --roi_center 45 51 29 --wm_frf_mask wm_frf_mask.nii.gz --gm_frf_mask gm_frf_mask.nii.gz --csf_frf_mask csf_frf_mask.nii.gz -f`
> 
> `scil_compute_memsmt_fodf.py wm_frf.txt gm_frf.txt csf_frf.txt --in_dwis dwi_lin_eddy_corrected_n4.nii.gz dwi_sph_eddy_corrected_n4.nii.gz --in_bvals dwi_lin.bval dwi_sph.bval --in_bvecs dwi_lin_eddy_corrected.eddy_rotated_bvecs dwi_sph_eddy_corrected.eddy_rotated_bvecs --in_bdeltas 1 0 --mask mean_b0_lin_topup_bet_mask.nii.gz --processes 8 -f`
> 
> `scil_compute_divide.py --in_dwis dwi_lin_eddy_corrected_n4.nii.gz dwi_sph_eddy_corrected_n4.nii.gz --in_bvals dwi_lin.bval dwi_sph.bval --in_bvecs dwi_lin_eddy_corrected.eddy_rotated_bvecs dwi_sph_eddy_corrected.eddy_rotated_bvecs --in_bdeltas 1 0 --mask mean_b0_lin_topup_bet_mask.nii.gz --fa fa.nii.gz --processes 8 -f`
> 
> Thanks!

Here is the data.

arnaudbore commented:
Build passed ! Good Job :beers: !

arnaudbore commented:
Build passed ! Good Job :beers: !

mdesco commented on scripts/scil_compute_memsmt_fodf.py:
```diff
@@ -0,0 +1,375 @@
+
+
+if __name__ == "__main__":
+    main()
```
/Users/desm2239/my_env/scilpy_p10/lib/python3.10/site-packages/cvxpy/problems/problem.py:1385: UserWarning: Solution may be inaccurate. Try another solver, adjusting the solver settings, or solve with verbose=True for more information.
  warnings.warn(

mdesco commented on scripts/scil_compute_divide.py:
```diff
@@ -0,0 +1,226 @@
+
+
+if __name__ == "__main__":
+    main()
```
(CVXPY) Mar 16 03:22:46 PM: Encountered unexpected exception importing solver SCS:
ImportError('dlopen(/Users/desm2239/my_env/scilpy_p10/lib/python3.10/site-packages/_scs_direct.cpython-310-darwin.so, 0x0002): Symbol not found: ___emutls_get_address\n  Referenced from: <17C4D491-3B6D-344A-8328-189690282A3A> /Users/desm2239/my_env/scilpy_p10/lib/python3.10/site-packages/scs/.dylibs/libopenblasp-r0.3.21.dylib\n  Expected in:     <E0C76B2B-0EC6-364E-AACD-42C7AAE73F23> /Applications/freesurfer/lib/gcc/lib/libgomp.1.dylib')

mdesco reviewed (commented):
Minor suggestions for @karanphil 
All the CVX errors seem like warnings, the output is created and it looks fine.

karanphil commented on docs/source/documentation/btensor_scripts.rst:
```diff
@@ -0,0 +1,18 @@
+If you want to do DIVIDE with b-tensor data, you should use the following command. It will save files for the MD, uFA, OP, MK_I, MK_A and MK_T. This script should run in about 1-2 hours for a full brain.
+::
+
+    scil_compute_divide.py --in_dwis dwi_linear.nii.gz dwi_planar.nii.gz dwi_spherical.nii.gz --in_bvals dwi_linear.bval dwi_planar.bval dwi_spherical.bval --in_bvecs dwi_linear.bvec dwi_planar.bvec dwi_spherical.bvec --in_bdeltas 1 -0.5 0 --mask mask.nii.gz --fa fa.nii.gz --processes 8 -f
```
I don't have the planar data, this example was very general.

karanphil commented on scripts/scil_compute_memsmt_frf.py:
```diff
@@ -0,0 +1,304 @@
+
+
+if __name__ == "__main__":
+    main()
```
I never got this error with Linux. This is weird. @arnaudbore what do you think? It looks like cvxpy tries to use some package that Max doesn't have.

arnaudbore commented:
Build passed ! Good Job :beers: !

mdesco reviewed (approved):
Tested on Laurent's mac. The errors and warnings are not there. 
Output are the same as mine. So, correct the small edits and good to merge. 

Good job!
